utils/src/bitmap/mod.rs

```diff
@@ -0,0 +1,1705 @@
+    pub fn new() -> Self {
+        // Invariant: chunks is never empty
+        let bitmap = VecDeque::from([Self::EMPTY_CHUNK]);
+        Self {
+            chunks: bitmap,
+            next_bit: 0,
+        }
+    }
+
+    // Create a new empty bitmap with the capacity to hold `size` bits without reallocating.
+    pub fn with_capacity(size: u64) -> Self {
+        let num_chunks = Self::num_chunks_at_size(size).get();
+        let mut bitmap = VecDeque::with_capacity(num_chunks);
+        // Invariant: chunks is never empty
+        bitmap.push_back(Self::EMPTY_CHUNK);
+        Self {
+            chunks: bitmap,
+            next_bit: 0,
+        }
+    }
+
+    // Returns the number of chunks in a bitmap with `size` bits.
+    fn num_chunks_at_size(size: u64) -> NonZeroUsize {
+        NZUsize!((size / Self::CHUNK_SIZE_BITS) as usize + 1)
+    }
+
+    /// Create a new bitmap with `size` bits, with all bits set to 0.
+    pub fn zeroes(size: u64) -> Self {
+        let num_chunks = Self::num_chunks_at_size(size).get();
+        let mut chunks = VecDeque::with_capacity(num_chunks);
+        for _ in 0..num_chunks {
+            chunks.push_back(Self::EMPTY_CHUNK);
+        }
+        Self {
+            chunks,
+            next_bit: size % Self::CHUNK_SIZE_BITS,
+        }
+    }
+
+    /// Create a new bitmap with `size` bits, with all bits set to 1.
+    pub fn ones(size: u64) -> Self {
+        let num_chunks = Self::num_chunks_at_size(size).get();
+        let mut chunks = VecDeque::with_capacity(num_chunks);
+        for _ in 0..num_chunks {
+            chunks.push_back(Self::FULL_CHUNK);
+        }
+        Self {
+            chunks,
+            next_bit: size % Self::CHUNK_SIZE_BITS,
+        }
+    }
+
+    /// Return the number of bits currently stored in the bitmap.
+    #[inline]
+    pub fn len(&self) -> u64 {
+        (self.chunks.len() as u64 - 1) * Self::CHUNK_SIZE_BITS + self.next_bit
+    }
+
+    /// Return the last chunk of the bitmap and its size in bits.
+    /// The size can be 0 (meaning the last chunk is empty).
+    #[inline]
+    fn last_chunk(&self) -> (&[u8; N], u64) {
+        (self.chunks.back().unwrap(), self.next_bit)
+    }
+
+    /// Return the last chunk of the bitmap and its size in bits.
+    /// The size can be 0 (meaning the last chunk is empty).
+    #[inline]
+    fn last_chunk_mut(&mut self) -> &mut [u8; N] {
+        self.chunks.back_mut().unwrap()
+    }
+
+    /// Returns the bitmap chunk containing the specified bit.
+    ///
+    /// # Warning
+    ///
+    /// Panics if the bit doesn't exist.
+    #[inline]
+    fn get_chunk(&self, bit_offset: u64) -> &[u8; N] {
+        assert!(bit_offset < self.len(), "out of bounds: {bit_offset}");
+        &self.chunks[Self::chunk_index(bit_offset)]
+    }
+
+    /// Get the value of a bit.
+    ///
+    /// # Warning
+    ///
+    /// Panics if the bit doesn't exist.
+    #[inline]
+    pub fn get_bit(&self, bit_offset: u64) -> bool {
```

Comment:
For public interface I would just name this `get` as you shouldn't have to publicly deal with chunks in the interface (?) Also because `set` is not `set_bit`